### PR TITLE
Added support for using the root certificates of the os

### PR DESCRIPTION
**Pull request by DavyLandman**
Sometimes, company firewalls like to inspect https traffic. To do that they insersect https traffic, and re-sign the traffic with a private certificate authority.

This code makes sure we import local certificate authorities.

### CODE REVIEW

File: rascal-vscode-extension/src/auto-jvm/JavaLookup.ts [hunk 1]
```diff
@@ -102,7 +102,7 @@ function getJavaCandidates(): string[] {
             }
         }
     }
-    return result;
+    return [];//result;
```
Comment (jurgenvinju) [inline rascal-vscode-extension/src/auto-jvm/JavaLookup.ts:105]: is this right?

File: rascal-vscode-extension/package.json [hunk 2]
```diff
@@ -20,9 +20,11 @@
     "url": "https://github.com/usethesource/rascal-language-servers"
   },
   "dependencies": {
+    "mac-ca": "^1.0.6",
```
Comment (jurgenvinju) [inline rascal-vscode-extension/package.json:23]: dependency added but not used?

File: rascal-vscode-extension/src/auto-jvm/typeles-ca.d.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// help typescript accept that it's all "any"
+declare module 'win-ca';
+declare module 'mac-ca';
```
Comment (jurgenvinju) [inline rascal-vscode-extension/src/auto-jvm/typeles-ca.d.ts:30]: don't see where this is used yet. pls help

**Review by jurgenvinju — COMMENTED** [review]
ok. I've some questions. see review. Also want to learn from this what to do in the Java code, but couldn't see that yet.

File: rascal-vscode-extension/src/auto-jvm/JavaLookup.ts (hunk 1 above)
Comment (DavyLandman) [reply rascal-vscode-extension/src/auto-jvm/JavaLookup.ts:105]: nope, this is debugging code, I wanted to trigger it on my machine. good find.

File: rascal-vscode-extension/package.json (hunk 2 above)
Comment (DavyLandman) [reply rascal-vscode-extension/package.json:23]: it's imported, that's enough.

see the comment:

```
import * as macca from 'mac-ca'; // just importing is is enough
```

File: rascal-vscode-extension/src/auto-jvm/typeles-ca.d.ts (hunk 3 above)
Comment (DavyLandman) [reply rascal-vscode-extension/src/auto-jvm/typeles-ca.d.ts:30]: well, typescript doesn't want to import modules without type definitions. but you can add a defintion for a module that roughly says: stop type-checking, everything is `any`. This file does that.

**Comment by DavyLandman** [comment]
Thanks for the review, I'm waiting for some feedback on this downstream issue: https://github.com/ukoloff/win-ca/issues/48

after that it either fixed or closed, we can choose to do this. else the downloader will sadly fail on any https-mitm-firewall-enterprise environments.

**Comment by jurgenvinju** [comment]
Ok that's great. I need to learn what to do for Mac by reading the code of the import you did. Let's merge this branch asap.

**Comment by DavyLandman** [comment]
I think we should move to this library: https://github.com/microsoft/vscode-windows-ca-certs in combination with https://github.com/microsoft/vscode-proxy-agent

**Comment by DavyLandman** [comment]
After some discussion with VS developers, they told me that vscode is already patching all nodejs processes to import the root certificate. I've tested this by installing a custom CA root in my own machine. and indeed, it just works as you would expect.

I'm closing this issue. @jurgenvinju we need to ask the users that reported this to provide a bit more information about their corporate firewall setup.

**Pull request closed without merging**